operator/internal/manifests/internal/config/loki-runtime-config.yaml

```diff
@@ -54,6 +54,23 @@ overrides:
     {{- if $spec.QueryLimits.CardinalityLimit }}
     cardinality_limit: {{ $spec.QueryLimits.CardinalityLimit }}
     {{- end }}
+    {{- with $l.Blocked }}
+    blocked_queries:
+    {{- range $blockedQuery := . }}
+    - {{ if $blockedQuery.Pattern -}}
+      pattern: '{{ $blockedQuery.Pattern }}'
```

Comment:
Do we need to pass `Pattern` through a function that escapes `'` or `"` (depending on what we use in the output)?

Reply:
You might be right. After looking on the Loki source code it expects the pattern to be a valid compiling regex. if we leave single quotes in the output `loki-runtime.yaml` the only case IMHO that needs attention is using single quotes in the regex statement (e.g. `.*msg="this 'service' is bad.*"`).

Creating a proper custom resource YAML with properly quoted single quotes results in an error as of now, because when we use the `pattern` fields value we get the non-escaped form, e.g.

For:
```yaml
  limits:
    tenants:
      application:
        queries:
          blocked:
          - pattern: '.*msg="this ''service'' is bad.*'
            regex: true
            types:
            - metric
```
results in an invalid `loki-runtime.yaml`:
```yaml
overrides:
  application:
    query_timeout: 3m
    blocked_queries:
    - pattern: '.*msg="this 'service' is bad.*'
      regex: true
      types: metric
```
and the pods get stuck with the following error:
```
level=error ts=2023-11-27T09:45:38.019488231Z caller=loki.go:517 msg="module failed" module=runtime-config error="invalid service state: Failed, expected: Running, failure: failed to load runtime config: unmarshal file \"/etc/loki/config/runtime-config.yaml\": yaml: line 5: did not find expected key"
```

WDYT?

Reply:
@xperimental Do you think we should render the regex pattern field's value as YAML literal scalar to escape the above? Something like this:
```
    {{- range $blockedQuery := . }}
    - {{ with $blockedQuery.Pattern -}}
      pattern: |
          {{ . }}`
      {{- end }}
```

Reply:
The block literal is probably a good idea :+1: 

We still need to process `Pattern` a bit, because it _might_ contain new-lines which we then need to treat specially. I have put comments in the different places to address this (`yamlBlock`).